### PR TITLE
BEI- 25: Create the comments table

### PR DESCRIPTION
**Pull request by hizikex**
#### WHY THIS PR?
-We need a database to store all comments, who made the comment and on which recipe.
-We want users to review a recipe and be able to comment on it.

#### WHAT DOES THIS PR?
It creates a table where users can comment what they feel about a recipe.

##### HOW CAN THIS BE TESTED
N/A

#### THINGS YOU MAY WANT THE REVIEWER TO NOTE
N/A

### CODE REVIEW

File: src/core/database/migrations/20230729164352-create-comments.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+'use strict';
+/** @type {import('sequelize-cli').Migration} */
```
Comment (LucianaDera) [inline src/core/database/migrations/20230729164352-create-comments.js:2]: Does this comment have any effect on the comment table

File: src/core/models/comments.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    targetKey: 'id',
+    as: 'users',
+  });
+  
+  Comment.belongsTo(Recipe, {
+    foreignKey: 'recipe_id',
+    targetKey: 'id',
+    as: 'recipe',
+  });
+
```
Comment (LucianaDera) [inline src/core/models/comments.ts:50]: please add an extra line

**Review by LucianaDera — CHANGES_REQUESTED** [review]

**Review by RICHYDOS — CHANGES_REQUESTED** [review]
All the stuff relating to the followers table should not be in this pull request

File: src/core/database/migrations/20230729164352-create-comments.js (hunk 1 above)
Comment (hizikex) [reply src/core/database/migrations/20230729164352-create-comments.js:2]: I don't think so...
Are you suggesting, I remove it

**Comment by hizikex** [comment]
> All the stuff relating to the followers table should not be in this pull request

I am hoping that followers will be merged first...

File: src/core/models/users.ts [hunk 3]
```diff
@@ -1,67 +1,68 @@
 import { Model, DataTypes, CreationOptional, InferCreationAttributes, InferAttributes } from 'sequelize';
 import { sequelize } from '../database/sequelize';
 import { UserAttributes } from '../interfaces/user';
+import { Follower } from './followers';
```
Comment (RICHYDOS) [inline src/core/models/users.ts:4]: Please remove this line because it doesn't have anything to do with this PR

**Review by RICHYDOS — CHANGES_REQUESTED** [review]

File: src/core/database/migrations/20230729164352-create-comments.js (hunk 1 above)
Comment (LucianaDera) [reply src/core/database/migrations/20230729164352-create-comments.js:2]: yes

**Review by RICHYDOS — APPROVED** [review]

File: src/core/database/migrations/20230729164352-create-comments.js (hunk 1 above)
Comment (LucianaDera) [reply src/core/database/migrations/20230729164352-create-comments.js:2]: yes

**Review by LucianaDera — CHANGES_REQUESTED** [review]

**Review by LucianaDera — APPROVED** [review]

**Pull request merged**